pkg/cmd/render/render.go

```diff
@@ -109,98 +134,206 @@ func runRenderCmd(cmd *cobra.Command, args []string) {
 	log.SetLevel(level)
 	log.Debug("debug logging enabled")
 
-	operatorDisabled := isDisabled()
+	err = render()
+	if err != nil {
+		log.WithError(err).Fatal("failed while rendering bootstrap assets")
+	}
+}
+
+func render() error {
+	operatorDisabledViaConfigmap := isDisabledViaConfigmap()
+
+	installConfigMode, err := getModeFromInstallConfig()
+	if err != nil {
+		return err
+	}
+	if !isValidMode(installConfigMode) {
+		return fmt.Errorf("invalid mode defined: %s", installConfigMode)
+	}
+
+	effectiveMode, conflict := utils.GetEffectiveOperatorMode(operatorDisabledViaConfigmap, installConfigMode)
+
+	if conflict {
+		return fmt.Errorf("config map asking for CCO to be disabled, and install-config asking for %q mode", installConfigMode)
+	}
 
 	log.Infof("Rendering files to %s", renderOpts.destinationDir)
 
 	ccoRenderDir := renderOpts.destinationDir
 
-	// render manifests
+	// render manifests from bindata
 	if err := os.MkdirAll(filepath.Join(ccoRenderDir, manifestsDir), 0775); err != nil {
-		log.WithError(err).Fatal("error creating manifests directory")
+		return errors.Wrap(err, "error creating manifests directory")
 	}
 
 	for _, assetName := range renderAssets {
 		asset, err := assets.Asset(assetName)
 		if err != nil {
-			log.WithError(err).Fatal("failed to read static asset")
+			return errors.Wrap(err, "failed to read static asset")
 		}
 
-		assetRenderPath := filepath.Join(ccoRenderDir, "manifests", "cco-"+filepath.Base(assetName))
-		log.Infof("Writing file: %s", assetRenderPath)
-		err = ioutil.WriteFile(assetRenderPath, asset, 0644)
-		if err != nil {
-			log.WithError(err).Fatal("failed to write file")
+		assetRenderPath := filepath.Join(ccoRenderDir, manifestsDir, "cco-"+filepath.Base(assetName))
+		if err := writeFile(assetRenderPath, asset); err != nil {
+			return err
 		}
 	}
 
+	// render operator config
+	var operatorConfig bytes.Buffer
+	templateVars := operatorTemplateVars{
+		CredentialsMode: effectiveMode,
+	}
+
+	if err := operatorConfigTemplate.Execute(&operatorConfig, templateVars); err != nil {
+		return errors.Wrap(err, "failed to execute operator config template")
+	}
+	assetRenderPath := filepath.Join(ccoRenderDir, manifestsDir, operatorConfigFilename)
+	if err := writeFile(assetRenderPath, operatorConfig.Bytes()); err != nil {
+		return err
+	}
+
 	// need at least the empty dir so the installer bootkube.sh script works as expected
 	if err := os.Mkdir(filepath.Join(ccoRenderDir, bootstrapManifestsDir), 0775); err != nil {
-		log.WithError(err).Fatal("error creating bootstrap-manifests directory")
+		errors.Wrap(err, "error creating bootstrap-manifests directory")
```

Comment:
This is not doing anything. I assume that it should be returning the wrapped error.

Reply:
Most definitely. Fixed.
I also slipped in some Makefile hacks to work around some issues I ran into regarding when the 'bindata' generation is happening.